flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/plan/nodes/exec/testutils/RestoreTestBase.java

```diff
@@ -0,0 +1,229 @@
+        final Path savepointPath = Paths.get(new URI(savepoint));
+        final Path savepointDirPath = getSavepointPath(program, getCurrentVersion());
+        Files.createDirectories(savepointDirPath);
+        Files.move(savepointPath, savepointDirPath, StandardCopyOption.ATOMIC_MOVE);
+    }
+
+    @ParameterizedTest
+    @MethodSource("createSpecs")
+    void testRestore(int version, TableTestProgram program) throws Exception {
+        final EnvironmentSettings settings = EnvironmentSettings.inStreamingMode();
+        final SavepointRestoreSettings restoreSettings =
+                SavepointRestoreSettings.forPath(
+                        getSavepointPath(program, version).toString(), false, RestoreMode.NO_CLAIM);
+        SavepointRestoreSettings.toConfiguration(restoreSettings, settings.getConfiguration());
+        final TableEnvironment tEnv = TableEnvironment.create(settings);
+        tEnv.getConfig()
+                .set(
+                        TableConfigOptions.PLAN_RESTORE_CATALOG_OBJECTS,
+                        TableConfigOptions.CatalogPlanRestore.IDENTIFIER);
+        for (SourceTestStep sourceTestStep : program.getSetupSourceTestSteps()) {
+            final String id = TestValuesTableFactory.registerData(sourceTestStep.dataAfterRestore);
+            final Map<String, String> options = new HashMap<>();
+            options.put("connector", "values");
+            options.put("data-id", id);
+            options.put("disable-lookup", "true");
+            options.put("runtime-source", "NewSource");
+            sourceTestStep.apply(tEnv, options);
+        }
+
+        for (SinkTestStep sinkTestStep : program.getSetupSinkTestSteps()) {
+            final Map<String, String> options = new HashMap<>();
+            options.put("connector", "values");
+            options.put("disable-lookup", "true");
+            options.put("sink-insert-only", "false");
+            sinkTestStep.apply(tEnv, options);
+        }
+
+        final CompiledPlan compiledPlan =
+                tEnv.loadPlan(PlanReference.fromFile(getPlanPath(program, version)));
+        compiledPlan.execute().await();
+        for (SinkTestStep sinkTestStep : program.getSetupSinkTestSteps()) {
+            assertThat(TestValuesTableFactory.getRawResults(sinkTestStep.name))
+                    .matches(l -> sinkTestStep.expectedAfterRestore.test((List<Row>) l));
+        }
+    }
+
+    private Path getPlanPath(TableTestProgram program, int version) {
+        return Paths.get(
+                getTestResourceDirectory(program, version) + "/plan/" + program.id + ".json");
+    }
+
+    private Path getSavepointPath(TableTestProgram program, int version) {
+        return Paths.get(getTestResourceDirectory(program, version) + "/savepoint/");
+    }
+
+    private String getTestResourceDirectory(TableTestProgram program, int version) {
+        return System.getProperty("user.dir")
```

Comment:
`user.dir`? shouldn't the logic here be similar to other plan tests do it? is this safe across all environments?

Reply:
I extracted this from https://github.com/apache/flink/blob/82fb74e23b1dfa46fa98c89a58d7f3126aeaec6c/flink-core/src/test/java/org/apache/flink/api/common/typeutils/TypeSerializerUpgradeTestBase.java#L503

Reply:
no, I don't. It is evaluated to the resources path